src/app/forms/components/dynamic-form-group/dynamic-form-group.component.html

```diff
@@ -55,7 +59,7 @@ <h1 class="govuk-heading-xl title">{{ form.meta.label }}</h1>
         [formControlName]="control.key"
         class="govuk-summary-list"
       ></div>
-      <app-dynamic-form-field *ngSwitchCase="true" [control]="control" [form]="formGroup"></app-dynamic-form-field>
+      <app-dynamic-form-field *ngSwitchCase="true" [control]="control" [form]="formGroup" [attr.id]="control.value.meta.id"></app-dynamic-form-field>
```

Comment:
Have you considered all the use cases for this `id`? When generating form arrays, there would be duplicate IDs in the DOM which should be avoided. Is having an id on the `app-dynamic-form-field` tag meaningful? The id would probably be more useful on the actual input/element with the value. Wick all that said. I think there are still instances of duplicate IDs due to the way dynamic forms are generated. Might be worth fixing that instead.